vault.py

```diff
@@ -503,74 +707,176 @@ class vdkPointBufferF64(Structure):
+
+class udQueryBoxFilter(udQueryFilter):
+
+  def __init__(self, position=[0,0,0], size=[1,1,1], yawPitchRoll=[0,0,0]):
+    super(udQueryBoxFilter, self).__init__()
+    self.__size = size
+    self.__yawPitchRoll = yawPitchRoll
+    self.position = position
+    self.size = size
+    self.yawPitchRoll = yawPitchRoll
+
+  @property
+  def position(self):
+    return self.__position
+
+  @position.setter
+  def position(self, position):
+    self.__position = tuple([*position])
+    self.SetAsBox(position, self.size, self.yawPitchRoll)
+
+  @property
+  def yawPitchRoll(self):
+    return self.__yawPitchRoll
+
+  @yawPitchRoll.setter
+  def yawPitchRoll(self, yawPitchRoll):
+    self.__yawPitchRoll = tuple([*yawPitchRoll])
+    self.SetAsBox(self.position, self.size, yawPitchRoll)
+
+  @property
+  def size(self):
+    return self.__size
+
+  @size.setter
+  def size(self, size):
+    self.__size = tuple([*size])
+    self.SetAsBox(self.position, [size[0]/2, size[1]/2, size[2]/2], self.yawPitchRoll)
+
+class udQueryContext:
+  def __init__(self, context: udContext, pointcloud: udPointCloud, filter: udQueryFilter):
+    self.udQueryContext_Create = getattr(udSDK, "udQueryContext_Create")
+    self.udQueryContext_ChangeFilter = getattr(udSDK, "udQueryContext_ChangeFilter")
+    self.udQueryContext_ChangePointCloud = getattr(udSDK, "udQueryContext_ChangePointCloud")
+    self.udQueryContext_ExecuteF64 = getattr(udSDK, "udQueryContext_ExecuteF64")
+    self.udQueryContext_ExecuteI64 = getattr(udSDK, "udQueryContext_ExecuteI64")
+    self.udQueryContext_Destroy = getattr(udSDK, "udQueryContext_Destroy")
+
+    self.context = context
+    self.pQueryContext = c_void_p(0)
+    self.pointcloud = pointcloud
+    self.filter = filter
+    self.Create()
+
+
+  def Create(self):
+    _HandleReturnValue(self.udQueryContext_Create(self.context.pContext, byref(self.pQueryContext), self.pointcloud.pPointCloud, self.filter.pFilter))
+
+  def ChangeFilter(self, filter: udQueryFilter):
+    self.filter = filter
+    _HandleReturnValue(self.udQueryContext_ChangeFilter(self.pQueryContext, filter.pFilter))
+
+  def ChangePointCloud(self, pointcloud: udPointCloud):
+    self.pointcloud = pointcloud
+    _HandleReturnValue(self.udQueryContext_ChangePointCloud(self.pQueryContext, pointcloud.pPointCloud))
+
+  def ExecuteF64(self, points: udPointBufferF64):
+    retVal = self.udQueryContext_ExecuteF64(self.pQueryContext, points)
+    if retVal == udError.NotFound:
+      return True
+    _HandleReturnValue(retVal)
+    return False
+
+  def ExecuteI64(self, points:udPointBufferI64):
+    retVal = self.udQueryContext_ExecuteI64(self.pQueryContext, points)
+    if retVal == udError.NotFound:
+      return True
+    _HandleReturnValue(retVal)
+    return False
+
+  def Destroy(self):
+    _HandleReturnValue(self.udQueryContext_Destroy(byref(self.pQueryContext)))
+
+class udStreamer(Structure):
+  _fields_ = [
+    ("active", c_bool),
```

Comment:
This isn't a bool, we explicitly removed `bool` from our API - if you come across any please report them.